php-checks/src/main/java/org/sonar/php/checks/security/AuthorizationsCheck.java

```diff
@@ -0,0 +1,201 @@
+import static org.sonar.plugins.php.api.tree.Tree.Kind.FUNCTION_EXPRESSION;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NULL_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.NUMERIC_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.REGULAR_STRING_LITERAL;
+import static org.sonar.plugins.php.api.tree.Tree.Kind.VARIABLE_IDENTIFIER;
+
+@Rule(key = "S5808")
+public class AuthorizationsCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Vote methods should return at least once a negative response";
+
+  private static final QualifiedName SYMFONY_VOTER_INTERFACE_NAMESPACE = qualifiedName("Symfony\\Component\\Security\\Core\\Authorization\\Voter\\VoterInterface");
+  private static final QualifiedName SYMFONY_VOTER_NAMESPACE = qualifiedName("Symfony\\Component\\Security\\Core\\Authorization\\Voter\\Voter");
+  private static final QualifiedName LARAVEL_GATE_NAMESPACE = qualifiedName("Illuminate\\Support\\Facades\\Gate");
+
+  private static final Set<String> VOTER_INTERFACE_COMPLIANT_RETURN_VALUES = new HashSet<>(Arrays.asList("ACCESS_ABSTAIN", "ACCESS_DENIED"));
+  private static final Set<String> LARAVEL_GATE_CLOSURE_COMPLIANT_RETURN_VALUES = new HashSet<>(Arrays.asList("false", "null"));
+
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree methodDeclarationTree) {
+    if (!hasModifier(methodDeclarationTree, "abstract")) {
+      String functionName = trimQuotes(getFunctionName(methodDeclarationTree));
+      if ("vote".equalsIgnoreCase(functionName) && isMethodInheritedFromClassOrInterface(SYMFONY_VOTER_INTERFACE_NAMESPACE, methodDeclarationTree)) {
+        checkReturnStatements(methodDeclarationTree, a -> (VOTER_INTERFACE_COMPLIANT_RETURN_VALUES.contains(a)));
+      }
+      if ("voteOnAttribute".equalsIgnoreCase(functionName) && isMethodInheritedFromClassOrInterface(SYMFONY_VOTER_NAMESPACE, methodDeclarationTree)) {
+        checkReturnStatements(methodDeclarationTree, a -> ("false".equals(a)));
+      }
+    }
+    super.visitMethodDeclaration(methodDeclarationTree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    ExpressionTree callee = tree.callee();
+    if (callee.is(CLASS_MEMBER_ACCESS)) {
+      MemberAccessTree memberAccessTree = (MemberAccessTree) callee;
+      Optional<CallArgumentTree> argument = Optional.empty();
+      if (isLaravelGateMethod(memberAccessTree, "define")) {
+        argument = argument(tree, "callback", 1);
+      }
+      if (isLaravelGateMethod(memberAccessTree, "before") || isLaravelGateMethod(memberAccessTree, "after")) {
+        argument = argument(tree, "callback", 0);
+      }
+      if (argument.isPresent()) {
+        ExpressionTree callArgumentTree = argument.get().value();
+        if (callArgumentTree.is(FUNCTION_EXPRESSION)) {
+          checkReturnStatements((FunctionExpressionTree) callArgumentTree, LARAVEL_GATE_CLOSURE_COMPLIANT_RETURN_VALUES::contains);
+        }
+      }
```

Comment:
```suggestion
argument
        .map(CallArgumentTree::value)
        .filter(FunctionExpressionTree.class::isInstance)
        .map(FunctionExpressionTree.class::cast)
        .ifPresent(a -> checkReturnStatements(a, LARAVEL_GATE_CLOSURE_COMPLIANT_RETURN_VALUES::contains));
```
I'm a fan of streams

Reply:
:D suggestion approved!